### PR TITLE
[Partner] Add langchain-google-genai package (gemini)

### PR DESCRIPTION
**Pull request by hinthornw**
Add a new ChatGoogleGenerativeAI class in a `langchain-google-genai` package.
Still todo: add a deprecation warning in PALM

### CODE REVIEW

File: libs/partners/google-genai/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,94 @@
+[tool.poetry]
+name = "langchain-google-genai"
+version = "0.0.1"
+description = "An integration package connecting Google's genai package and LangChain"
+authors = []
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = ">=3.9,<4.0"
```
Comment (hinthornw) [inline libs/partners/google-genai/pyproject.toml:9]: Had to increase the min version since google-generativeai doesn't permit python 3.8 fyi

**Review by efriis — APPROVED** [review]

File: libs/partners/google-genai/langchain_google_genai/chat_models.py [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+            )
+
+        values["client"] = genai
+        if (
+            values.get("temperature") is not None
+            and not 0 <= values["temperature"] <= 1
+        ):
+            raise ValueError("temperature must be in the range [0.0, 1.0]")
+
+        if values.get("top_p") is not None and not 0 <= values["top_p"] <= 1:
```
Comment (baskaryan) [inline libs/partners/google-genai/langchain_google_genai/chat_models.py:457]: i think top_p isn't specified as class attribute, should it be? also candidate_count

Comment (baskaryan) [reply libs/partners/google-genai/langchain_google_genai/chat_models.py:457]: oh see we've got `n` for candidate count

File: libs/partners/google-genai/langchain_google_genai/chat_models.py [hunk 3]
```diff
@@ -0,0 +1,556 @@
...
+        self, messages: Sequence[BaseMessage], stop: Optional[List[str]]
+    ) -> Dict[str, Any]:
+        contents = _messages_to_genai_contents(messages)
+        gen_config = {
+            k: v
+            for k, v in {
+                "candidate_count": self.n,
+                "temperature": self.temperature,
+                "stop_sequences": stop,
+                "max_output_tokens": self.max_output_tokens,
```
Comment (baskaryan) [inline libs/partners/google-genai/langchain_google_genai/chat_models.py:495]: should top_k and top_p be here?

File: libs/partners/google-genai/langchain_google_genai/chat_models.py [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+    ) -> ChatResult:
+        params = self._prepare_params(messages, stop)
+        response: genai.types.GenerateContentResponse = chat_with_retry(
+            **params,
+            generation_method=self._generation_method,
+            **kwargs,
+        )
+        return _response_to_result(response)
+
+    async def _agenerate(
```
Comment (baskaryan) [inline libs/partners/google-genai/langchain_google_genai/chat_models.py:520]: believe this is implemented by default

File: libs/partners/google-genai/langchain_google_genai/chat_models.py (hunk 2 above)
Comment (hinthornw) [reply libs/partners/google-genai/langchain_google_genai/chat_models.py:457]: Actually top_p is no longer supported, my fault

File: libs/partners/google-genai/langchain_google_genai/chat_models.py (hunk 3 above)
Comment (hinthornw) [reply libs/partners/google-genai/langchain_google_genai/chat_models.py:495]: Think only these are supported now
```
class GenerationConfig:
    candidate_count: int | None = 0
    stop_sequences: Iterable[str] | None = dataclasses.field(default_factory=list)
    max_output_tokens: int | None = 0
    temperature: float | None = 0.0
    ```

Comment (baskaryan) [reply libs/partners/google-genai/langchain_google_genai/chat_models.py:495]: Screenshot 2023-12-13 at 10 39 13 AM

https://ai.google.dev/api/python/google/generativeai/GenerationConfig

**Comment by hinthornw** [comment]
Thanks for fixing up CI!

**Pull request merged**